src/main/java/ch/uzh/ifi/fs22/sel/group15/cozycave/server/controller/ListingController.java

```diff
@@ -120,4 +129,165 @@ public void deleteListing(@PathVariable UUID id) {
 
         log.info("listing with id {} deleted", id);
     }
+
+    // get all applications of a listing
+    @GetMapping("/listings/{id}/applications")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public List<ApplicationGetDto> findApplications(
+            @AuthenticationPrincipal String authUserId,
+            @PathVariable UUID id) {
+        User authUser = userService.findUserID(UUID.fromString(authUserId))
+                .orElseThrow(() -> {
+                    log.error("user (authenticated user) with id {} not found while getting application", authUserId);
+                    return new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR,
+                            "error finding authenticated user");
+                });
+
+        Listing listing = listingService.findListingById(id)
+                .orElseThrow(() -> {
+                    log.debug("listing with id {} not found while geting listing", id);
+                    return new ResponseStatusException(HttpStatus.NOT_FOUND,
+                            "listing with id " + id + " not found");
+                });
+        if (authUser.getId() != listing.getPublisher().getId()) {
```

Comment:
this is ok for now but because I haven't added authorization checks yet, this could change later